src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -138,9 +139,14 @@ def sagemaker_train(train_config, data_config, train_path, val_path, model_dir,
     combine_train_val = '_kfold' in validated_train_config
     train_dmatrix, val_dmatrix, train_val_dmatrix = get_validated_dmatrices(train_path, val_path, file_type,
                                                                             csv_weights, is_pipe, combine_train_val)
-
     checkpoint_dir = checkpoint_config.get("LocalPath", None)
-
+    if val_path is not None:
+        if train_path == val_path or os.path.basename(train_path) == os.path.basename(val_path):
+            logger.warning('Found same path for training and validation. This is not recommended and results may not '
+                           'be correct.')
+        elif not is_pipe:
+            # Check if there is potential data redundancy between training and validation sets
+            check_data_redundancy(train_path, val_path)
```

Comment:
Is this correct? This is saying

1. If we suspect data and validation redundancy => log a warning
2. Else (we DON'T suspect data and validation redundancy) AND we're in file mode => check data redundancy

Shouldn't the call to `check_data_redundancy` fall under the same parent condition as the path check and the warning?

Reply:
Discussed with @haixiw 
The idea was:

1. If we KNOW that the data and validation paths are same, then log a warning (log a warning soon as we can)
2. Else(Shallow path check did not find redundancy and we want file level comparison) AND if we are in file mode , run the file-by-file comparison

We want to run `check_data_redundancy` only when necessary, that's why it's under "else". We don't want to run this file-by-file check on "pipe" mode that's why the condition.